Source/Atomic/UI/UI.cpp

```diff
@@ -797,6 +798,14 @@ UIWidget* UI::WrapWidget(tb::TBWidget* widget)
         return nwidget;
     }
 
+    if (widget->IsOfType<TBBarGraph>())
+    {
+        UIBargraph* nwidget = new UIBargraph(context_, false);
+        nwidget->SetWidget(widget);
+        WrapWidget(nwidget, widget);
```

Comment:
Awesome! 👍  Wrapping widgets is far less boiler plate than it used to be, though is still tricky due to needing the reverse type checks when wrapping here.  This is an area for UI subsystem improvement, could also be documented in an "adding new widgets" cookbook article, nice work :)